### PR TITLE
Purview data policies APIs specs initial

### PR DESCRIPTION
**Pull request by sbhat9**
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [x] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document: https://microsoftapc-my.sharepoint.com/:w:/g/personal/sagarbhat_microsoft_com/EdVHYU1uX2ZDjljONkxXb9kBlEy6UUvtDuQ3V1YuzqW5vw?e=ZkCbOu
* Previous API Spec Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

**Comment by inward-eye** [comment]
This PR is related to:
API review: https://apps.powerapps.com/play/e/ed2ffefd-774d-40dd-ab23-7fff01aeec9f/a/e41c69de-ef96-44ba-b287-0e012e7c13ae
Corresponding issue: https://github.com/Azure/azure-rest-api-specs-pr/issues/13514

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json [hunk 1]
```diff
@@ -0,0 +1,616 @@
...
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DataPolicyList"
+            }
+          },
+          "default": {
+            "description": "An error response received from the Policy service",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
```
Comment (mikekistler) [inline specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:34]: Don't cross the streams. Don't reference resource-management common types. Define your own error model either in this file or in a Purview (data-plane) common types file.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json [hunk 2]
```diff
@@ -0,0 +1,616 @@
...
+  },
+  "paths": {
+    "/data-policies": {
+      "get": {
+        "tags": [
+          "List_DataPolicy"
+        ],
+        "summary": "API to list the purview data policies",
+        "description": "List all data policies.",
+        "operationId": "List_DataPolicy",
```
Comment (mikekistler) [inline specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:15]: The Azure convention for operationIds is "noun_verb"
```suggestion
        "operationId": "DataPolicy_List",
```

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json [hunk 3]
```diff
@@ -0,0 +1,616 @@
...
+      "name": "policy-name",
+      "description": "Name of the policy to be deleted.",
+      "required": true,
+      "type": "string"
+    },
+    "if-match": {
+      "in": "header",
+      "name": "If-Match",
+      "description": "Http Etag value of a policy to be updated. Required only during the replace operation",
+      "type": "string"
```
Comment (mikekistler) [inline specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:585]: I think we may want `format: etag` for this header ... for languages like .NET that have a type for ETags.
```suggestion
      "type": "string",
      "format": "etag"
```

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json [hunk 4]
```diff
@@ -0,0 +1,616 @@
...
+        "description"
+      ],
+      "type": "object",
+      "description": "Purview data policy",
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The policy id"
+        },
```
Comment (mikekistler) [inline specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:371]: Not sure what this property is or why it is needed. It appears nowhere else in the API that I could find. Can you eliminate it?

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json [hunk 5]
```diff
@@ -0,0 +1,616 @@
...
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The policy name"
+        },
```
Comment (mikekistler) [inline specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:376]: I assume that name is just the "policyName" from the URL reflected into the resource. If so then this property should be required.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json [hunk 6]
```diff
@@ -0,0 +1,616 @@
...
+          "x-ms-identifiers": []
+        },
+        "aadMembers": {
+          "type": "array",
+          "description": "Array of azure active directory members",
+          "items": {
+            "$ref": "#/definitions/AADMember"
+          },
+          "x-ms-identifiers": []
+        }
```
Comment (mikekistler) [inline specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:459]: As a user I would scratch my head here: two different properties with different types but the exact same description. I think users will need more information for these to know what they mean / how to set them.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json [hunk 7]
```diff
@@ -0,0 +1,616 @@
...
+        },
+        "x-ms-examples": {
+          "DevopsPolicy_List": {
+            "$ref": "./examples/DataPolicy_List.json"
+          }
+        }
+      }
+    },
+    "/data-policies/{policy-name}": {
+      "put": {
```
Comment (mikekistler) [inline specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:56]: Azure guidelines prefer PATCH over PUT for resource create -- it is more version resilient.

**Review by mikekistler — CHANGES_REQUESTED** [review]
I flagged a bunch of minor issues that I hope you can address in this version.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 1 above)
Comment (weidongxu-microsoft) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:34]: One can actually use the shared data-plane types https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/data-plane/v1/types.json

Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json]: Used shared data-plane types as suggested.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 2 above)
Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json]: updated.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 4 above)
Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json]: removed. It is not needed.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json [hunk 8]
```diff
@@ -0,0 +1,616 @@
...
+        "kind": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The policy kind"
+        },
+        "source": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The policy source"
+        },
```
[the remark this answers is not in the document]
Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json]: Removed the properties.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 6 above)
Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json]: As decided during the review, removed the `fabricItemMembers` field.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 3 above)
Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:656]: If we change to format: etag. It is causing some validation errors. Keeping the type as string.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json [hunk 9]
```diff
@@ -0,0 +1,616 @@
...
+      "type": "string"
+    }
+  },
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "flow": "implicit",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
```
[the remark this answers is not in the document]
Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:665]: Can you please provide some example here? Most of our other APIs

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 5 above)
Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json]: updated.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (mikekistler) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:665]: A good example is [deviceupdate](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/deviceupdate/data-plane/Microsoft.DeviceUpdate/stable/2022-10-01/deviceupdate.json#L40).

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 3 above)
Comment (mikekistler) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:656]: What are the validation errors?

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json [hunk 10]
```diff
@@ -0,0 +1,687 @@
...
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DataPolicy"
+            },
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:252]: The `state` parameter seems to be the only difference compared to `/data-policies/{policyName}` 

What is the difference of this API, compared to request `/data-policies/{policyName}` then filter by `state`?

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json [hunk 11]
```diff
@@ -0,0 +1,687 @@
...
+      "type": "string"
+    },
+    "state": {
+      "in": "path",
+      "name": "state",
+      "description": "policy state",
+      "required": true,
+      "type": "string"
+    },
+    "if-match": {
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:652]: nit, prefer camelCase on ref name

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json [hunk 12]
```diff
@@ -0,0 +1,687 @@
...
+      "required": true,
+      "type": "string"
+    },
+    "state": {
+      "in": "path",
+      "name": "state",
+      "description": "policy state",
+      "required": true,
+      "type": "string"
+    },
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:651]: We may not need it, depends on https://github.com/Azure/azure-rest-api-specs/pull/25059#discussion_r1298063068

If we really need it, I assume it be extensible enum?

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json [hunk 13]
```diff
@@ -0,0 +1,687 @@
...
+        },
+        "scopes": {
+          "type": "array",
+          "readOnly": true,
+          "description": "Array of scopes where the policy is published",
+          "items": {
+            "type": "string"
+          },
+          "x-ms-identifiers": []
+        },
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:462]: nit, What is a `scope`? Is it ARM concept? I think better doc would help here.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json [hunk 14]
```diff
@@ -0,0 +1,687 @@
...
+        "itemAccess": {
+          "type": "array",
+          "description": "Array of access items for the member",
+          "items": {
+            "type": "string",
+            "description": "access item"
+          }
+        }
+      }
+    },
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:538]: Where is this used?

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json [hunk 15]
```diff
@@ -0,0 +1,687 @@
...
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/policyName"
+          },
+          {
+            "$ref": "#/parameters/if-match"
+          },
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:75]: Why it has only `if-match`?

E.g. If I want to create a resource but don't want it overwrite an existing one (in possible concurrency situation), I would like to use `if-none-match: *` here.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json [hunk 16]
```diff
@@ -0,0 +1,687 @@
...
+          "description": "values for a policy permission",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "SystemData": {
+      "type": "object",
+      "readOnly": true,
```
[the remark this answers is not in the document]
Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json]: added a description.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 10 above)
Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json]: Thanks for the feedback. We had two review meetings with the review team. It follows the similar pattern to the key vaults APIs(not exactly same). Below are the final set of APIs after the review. 
image

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 11 above)
Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json]: updated.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 12 above)
Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:634]: updated the above comment with details.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 13 above)
Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json]: updated the description.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 14 above)
Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json]: removed.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 15 above)
Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json]: In purview policy we only support create and full replace of a policy. We do not support `if-none-match: *` in this version.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json]: Thanks, updated now.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json [hunk 17]
```diff
@@ -0,0 +1,670 @@
...
+            "$ref": "#/definitions/PolicyDecisionRule"
+          }
+        }
+      }
+    },
+    "PolicyMembers": {
+      "type": "object",
+      "description": "Policy member",
+      "properties": {
+        "entraMembers": {
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:484]: What is "entra"?

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 10 above)
Comment (weidongxu-microsoft) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json]: OK. I was in the meeting. So if Mike approves it, I am good.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 17 above)
Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:484]: Azure active directory is renamed to `entra`. So, PMs are asking us to rename `aadMembers` field as `entraMembers`.
@inward-eye can you please elaborate?

Comment (weidongxu-microsoft) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:484]: I see. Thanks.
https://learn.microsoft.com/en-us/azure/active-directory/fundamentals/new-name

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 3 above)
Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:639]: https://github.com/Azure/azure-rest-api-specs/runs/15950819967

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 7 above)
Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:56]: In purview policy space we are currently supporting only the create and replace. Partial update of a policy is not supported.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 3 above)
Comment (weidongxu-microsoft) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:639]: Its 

```
type: string,
format: etag
```

not
```
type: etag
```

Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:645]: It now gives below error. "etag is not a valid format"
https://github.com/Azure/azure-rest-api-specs/pull/25059/checks?check_run_id=16064919893

Comment (mikekistler) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:645]: Sorry, we were misinformed on this one. `etag` is not (currently) a format recognized by autorest or the openapi-validator. So drop the `format: etag` for now.

**Review by mikekistler — APPROVED** [review]
Looks good. 👍 

You can remove the `format: etag` to fix the lint error, but also fine as is.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json (hunk 3 above)
Comment (sbhat9) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json:639]: thanks.

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by mikekistler** [comment]
@sbhat9 Do you still want to go forward with these changes? If so please re-open the PR and I'll mark it as approved when you say it's ready to merge.

**Pull request closed without merging**